File: src/main/java/com/example/store/controller/UserController.java
package com.example.store.controller;

import com.example.store.dto.ResponseMessage;
import com.example.store.dto.UserRequest;
import com.example.store.service.IUserService;
import lombok.AllArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@AllArgsConstructor
@Slf4j
@RequestMapping("/api/v1/users/")
public class UserController {

    private IUserService userService;

    @PostMapping("user")
    public ResponseEntity<ResponseMessage> save(@RequestBody UserRequest userRequest) {
        userService.save(userRequest);
        return new ResponseEntity(new ResponseMessage(HttpStatus.OK.value(), "User Saved Successfully" ), HttpStatus.OK);
    }
}

File: src/main/kotlin/com/example/store/controller/UserController.kt
package com.example.store.controller

import com.example.store.dto.ResponseMessage
import com.example.store.dto.UserRequest
import com.example.store.service.IUserService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v1/users/")
class UserController {
    @Autowired
    private lateinit var userService: IUserService

    @PostMapping("user")
    fun save(@RequestBody userRequest: UserRequest?): Any? {
        userService?.save(userRequest)
        return ResponseEntity<Any?>(ResponseMessage(HttpStatus.OK.value(), "User Saved Successfully"), HttpStatus.OK)
    }
}